src/Arcus.Observability.Telemetry.Sql/SqlConnectionStringParserResult.cs

```diff
@@ -0,0 +1,94 @@
+﻿using GuardNet;
+
+namespace Arcus.Observability.Telemetry.Sql
+{
+    /// <summary>
+    /// Represents the result of the <see cref="SqlConnectionStringParser"/> when parsing a SQL connection string.
+    /// </summary>
+    public class SqlConnectionStringParserResult
```

Comment:
I agree that we should move this type to the core package, and remove the 'Sql' package.
I even think that this type can be internal ?

Reply:
Could be internal, but then we wouldn't be able to test it anymore; while, as you can see, there are a lot of tests for this.

Reply:
I use `InternalsVisibleTo` for these kind of things.

Reply:
I know, it's just that this often feels like an anti-pattern or a sign of bad design, for me.

Reply:
I think it is just a pragmatic way to make sure that you can test functionality that can be kept internal.  I have more problems with changing the visibility of a type to public, just to make it testable :)

Reply:
Yeah, that's a different way of looking at things. I'm used to working with a test-driven, type-driven approach where the test is the only place first where your code is being used, and so, is already public. This was also the case here. We could make it internal but then all the tests have to be changed as well. 'Just to make it testable' is something that is often the argument of non-test-driven approaches, I guess.
I'll see if we can change this.

Reply:
Made sure that the parsing is moved to the `.Core` package, made the parsing logic internal while still test the parsing via the logging extensions instead so that we do not need the `InternalsVisibleTo`.

Reply:
Great thinking!